FogCreek/Glitch-Community
# Boiling sunflower: Hide suspended project details

## Links
* Remix link: https://boiling-sunflower.glitch.me/
* Manuscript link: https://glitch.fogbugz.com/f/cases/edit/3328294/Suspended-projects-shouldn-t-display-a-standard-project-page
* Specs/Notion docs: n/a

## GIF/Screenshots:
![Screenshot 2019-05-03 at 11 33 12 AM](https://user-images.githubusercontent.com/1237410/57158069-4cf31880-6d97-11e9-8710-16e89ed5c94e.png)

## Changes:
* Project pages for suspended projects will show "suspended project" for the project domain and description, and a "no entry" emoji image for the project avatar to non-owners.
* Project owners will still see the information for a suspended project as normal, in case they need to change it.
* `<ProfileContainer type="team|user|project" />` has been replaced with specific profile containers for each type, and have been moved into `@Components/containers/profile` with an index to allow importing them at that level. (Like `import { ProjectProfileContainer } from @Components/containers/profile;`)
* `TrackedButton` was added to buttons and allows the same props as `Button` to be passed through, to wrap the tracking function.

## How To Test:
* Check out https://boiling-sunflower.glitch.me/~a-bad-project and confirm that you only see "suspended project" and the "no entry" emoji sign
* Make sure avatar/cover actions still work as expected for this

## Feedback I'm looking for:
- I think we could just always pass `avatarActions` and `coverActions` to these now with `isAuthorized` and conditionally show them based on that, rather than doing a map/filter (which is repeated a lot)
- Is there anything about `TrackedButton` that could be improved? Would people prefer `trackingLabel` over `label` or something else?

## Things left to do before deploying:
- [x] Remove type props from places

💁 We should follow up on an editor PR that shows the suspended project pop-up, since it still displays the name. (Replace it with "This project has been suspended")

💁 This also gets put into OpenGraph data for unfurls. We can do a separate PR to hide that as well if we want to.
